app/models/job.py

```diff
@@ -108,13 +108,20 @@ def finished_processing(self):
         return self.notification_count == self.notifications_sent
 
     @property
-    def awaiting_processing_or_recently_processed(self):
-        if not self.processing_started:
+    def awaiting_processing_or_recently_processed_message(self):
+        data_retention = self.service_data_retention_days
```

Comment:
There’s already a method for this on the service object which accounts for different retention periods per channel:
```suggestion
        data_retention = self.service.get_days_of_retention(self.template_type) 
```

Reply:
I was aware of this, I had tried to handle this within the `Job` model and complicated things unnecessarily .
I am now taking a different straightforward approach which does not need an adjustment to `awaiting_processing_or_recently_processed`